modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -248,6 +388,47 @@ The operation generates a foreground mask.
 */
 GAPI_EXPORTS GMat BackgroundSubtractor(const GMat& src, const cv::gapi::video::BackgroundSubtractorParams& bsParams);
 
+/** @brief Standard Kalman filter algorithm. The operation uses standard matrices
+by default <http://en.wikipedia.org/wiki/Kalman_filter>.
+transitionMatrix, controlMatrix and measurementMatrix can be modified to get an
+extended Kalman filter functionality. However, according to OCV's Kalman filter
+implementation transitionMatrix, processNoiseCov and measurementNoiseCov shouldn't
+be set to zero as it'll lead unexpected behavior.
+
+@return Output image is predicted or corrected state, i.e. 32-bit or 64-bit float
+1-channel matrix @ref CV_32FC1 or @ref CV_64FC1.
+If measurement matrix is given (haveMeasurements == true), corrected state will
+be returned which corresponds to the pipeline
+cv::KalmanFilter::predict(control) -> cv::KalmanFilter::correct(measurement).
+Otherwise, predicted state will be returned which corresponds to the call of
+cv::KalmanFilter::predict(control).
+@note Functional textual ID is "org.opencv.video.KalmanFilter"
+
+@param measurement input matrix: 32-bit or 64-bit float 1-channel matrix contains measuremens.
+@param haveMeasurement dynamic input flag that indicates whether we get measurements
+at a particular iteration .
+@param control input matrix: 32-bit or 64-bit float 1-channel matrix contains control data
+for changing dynamic system.
+@param kfParams Set of initialization parameters for Kalman filter kernel.
+*/
+GAPI_EXPORTS GMat KalmanFilter(const GMat& measurement, const GOpaque<bool>& haveMeasurement,
+                               const GMat& control, const cv::gapi::video::KalmanParams& kfParams);
+
+/** @overload
+Standard Kalman filter algorithm.
+
+@return Output image is predicted or corrected state, i.e. 32-bit or 64-bit float
+1-channel matrix @ref CV_32FC1 or @ref CV_64FC1.
+@note Function textual ID is "org.opencv.video.KalmanFilterNoControl"
+
+@param measurement input matrix: 32-bit or 64-bit float 1-channel matrix contains measuremens.
+@param haveMeasurement dynamic input flag that indicates whether we get measurements
+at a particular iteration.
+@param kfParams Set of initialization parameters for Kalman filter kernel.
+ */
+GAPI_EXPORTS GMat KalmanFilter(const GMat& measurement, const GOpaque<bool>& haveMeasurement,
+                               const cv::gapi::video::KalmanParams& kfParams);
```

Comment:
Please add some words on this overload has no control

Reply:
Done.